solutions/src/main/java/ru/sberbank/school/task06/MapCounter.java

```diff
@@ -0,0 +1,67 @@
+package ru.sberbank.school.task06;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Created by Mart
+ * 25.05.2019
+ **/
+public class MapCounter <K> implements CountMap {
+    private Map <K, Integer> map = new HashMap<>();
+
+    @Override
+    public void add(Object o) {
+        if (map.containsKey(o)) {
+           map.put((K) o, map.get(o) + 1);
+        } else {
+            map.put((K) o, 1);
+        }
+    }
+
+    @Override
+    public int getCount(Object o) {
+        return map.get(o);
+    }
+
+    @Override
+    public int remove(Object o) {
+        int count = map.get(o);
+        map.remove(o);
+        return count;
+    }
+
+    @Override
+    public int size() {
+        return map.size();
+    }
+
+    @Override
+    public void addAll(CountMap source) {
+        Map sourceMap = source.toMap();
+        for (Object o : sourceMap.keySet()) {
+            this.add(o, (Integer) sourceMap.get(o));
+        }
+    }
+
+    @Override
+    public Map<? extends K, Integer> toMap() {
+        return map;
+    }
+
+    @Override
+    public void toMap(Map destination) {
+        MapCounter destMap = (MapCounter) destination;
+        for (Object o : map.keySet()) {
+            destMap.add(o, map.get(o));
+        }
+    }
+
+    private void add(Object o, int counter) {
```

Comment:
Overload methods should not be split. Previous overloaded method located at line &#39;13&#39;.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1820874637>*